[Buildkite] Test Android Staging on `ami-0953ca1d949c23a58` -> `ami-0f976e0b4de96aa28`

ParaskP7 opened the pull request:
Related: [a8c-ci-toolkit-buildkite-plugin##142
](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/142)

AMI Name: `android-build-image-6.12.0v1.5-rc-1`

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR changes the Buildkite agent from `android` to `android-staging`. This change is not meant to be merged, but rather used to verify that the new AMI `ami-0953ca1d949c23a58` works as expected.

---

### Before the Change

---

- [Saving Gradle dependency cache...](https://buildkite.com/automattic/wordpress-android/builds/20909#01947bdd-80e0-4097-922d-842f5b65ef41/460-461)
  - Duration: [0 minutes and 55 seconds](https://buildkite.com/automattic/wordpress-android/builds/20909#01947bdd-80e0-4097-922d-842f5b65ef41/460-471)
  - Cache Size: [1076.73 MB](https://buildkite.com/automattic/wordpress-android/builds/20909#01947bdd-80e0-4097-922d-842f5b65ef41/460-472)
- Build Time: [22m 59s](https://buildkite.com/automattic/wordpress-android/builds/20909#01947bdd-80e0-4097-922d-842f5b65ef41)

-----

- [Restoring Gradle dependency cache...](https://buildkite.com/automattic/wordpress-android/builds/20961#019497f2-ab5f-4014-a5a3-d40af06de47e/192-193)
  - Duration: [0 minutes and 22 seconds](https://buildkite.com/automattic/wordpress-android/builds/20961#019497f2-ab5f-4014-a5a3-d40af06de47e/192-201)
  - Cache Size: [1076.73 MB](https://buildkite.com/automattic/wordpress-android/builds/20961#019497f2-ab5f-4014-a5a3-d40af06de47e/192-202)
- [Build](https://buildkite.com/automattic/wordpress-android/builds/20961#019497f2-ab5f-4014-a5a3-d40af06de47e) (`Lint`): [Build Scan](https://buildkite.com/automattic/wordpress-android/builds/20961#019497f2-ab5f-4014-a5a3-d40af06de47e/259-391) -> [Network Activity*](https://scans.gradle.com/s/yki6krscgr7fo/performance/network-activity)
  - Files Downloaded: `54`
  - Network Requests: `79`
  - Time Spent: `5s`

(*) Although, note that these numbers might not be very representative because `trunk` is using a dependency cache that is already outdated (5 days has passed since it was last updated, every Sunday).

---

### After the Change

---

- [Saving Gradle dependency cache...](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43/23409-23410)
  - Duration: [1 minutes and 40 seconds](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43/23409-23420)
  - Cache Size: [1563.65 MB](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43/23409-23421)
- Build Time: [8m 29s](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43)

-----

- [Restoring Gradle dependency cache...](https://buildkite.com/automattic/wordpress-android/builds/20968#019498fd-4d26-4c23-a6bd-74d4d0387846/194-195)
  - Duration: [0 minutes and 33 seconds](https://buildkite.com/automattic/wordpress-android/builds/20968#019498fd-4d26-4c23-a6bd-74d4d0387846/194-203)
  - Cache Size: [1563.65 MB](https://buildkite.com/automattic/wordpress-android/builds/20968#019498fd-4d26-4c23-a6bd-74d4d0387846/194-204)
- [Build](https://buildkite.com/automattic/wordpress-android/builds/20968#019498fd-4d26-4c23-a6bd-74d4d0387846) (`Lint`): [Build Scan](https://buildkite.com/automattic/wordpress-android/builds/20968#019498fd-4d26-4c23-a6bd-74d4d0387846/261-330) -> [Network Activity](https://scans.gradle.com/s/2c6q4zhfmwheo/performance/network-activity)
  - Files Downloaded: `0`
  - Network Requests: `0`
  - Time Spent: `0`

---

### Conclusions

As expected, using `./gradlew downloadDependencies androidDependencies` is:
1. Simpler, much less configuration and reasoning about which tasks to use.
2. Scalable, no need to add any more tasks to the `save-cache.sh` script as the project evolves.
3. Producing a bigger cache,
    - Before: [1076.73 MB](https://buildkite.com/automattic/wordpress-android/builds/20909#01947bdd-80e0-4097-922d-842f5b65ef41/460-472)
    - After: [1563.65 MB](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43/23409-23421)
    - Result: `+ 486.92 MB` more data downloaded, which means even less network requests on CI.
4. Slower, both in saving and restoring the cache, which is the only drawable:
    - Save:
        - Before: [0 minutes and 55 seconds](https://buildkite.com/automattic/wordpress-android/builds/20909#01947bdd-80e0-4097-922d-842f5b65ef41/460-471)
        - After: [1 minutes and 40 seconds](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43/23409-23420)
        - Result: `+ 45 seconds` to save the cache.
    - Restore:
        - Before: [0 minutes and 22 seconds](https://buildkite.com/automattic/wordpress-android/builds/20961#019497f2-ab5f-4014-a5a3-d40af06de47e/192-201)
        - After: [0 minutes and 33 seconds](https://buildkite.com/automattic/wordpress-android/builds/20968#019498fd-4d26-4c23-a6bd-74d4d0387846/194-203)
        - Result: `+ 11 seconds` to restore the cache.

ParaskP7 committed:
CI: Use android-staging agent to test ami-0953ca1d949c23a58

AMI Name: android-build-image-6.12.0v1.5-rc-1

ParaskP7 committed:
CI: Update a8c-ci-toolkit to pr-sha commit (test download dependencies)

FYI: This change is done for testing purposes only, and in order to
verify that the newly added 'init.gradle.kts' script is working as
expected.

A8C CI Toolkit PR: [Dependency Cache] Test Custom Init Gradle Task
to Download Dependencies #142
- Automattic/a8c-ci-toolkit-buildkite-plugin#142

ParaskP7 committed:
CI: Replace all individual assemble tasks with download dependencies

This custom 'downloadDependencies' task is part of the Android AMI,
that is utilized by the 'android' agents, and it is becoming available
to all project via the 'init.gradle.kts' script that is included in it.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 commented:
👋 @wzieba same as with [WCAndroid](https://github.com/woocommerce/woocommerce-android/pull/13387#issuecomment-2612631091), please take a look! 🙏

ParaskP7 committed:
CI: Remove unnecessary android dependencies from download dependencies

A new Android AMI got created that now finalizes 'downloadDependencies'
with the 'androidDependencies' task.

ParaskP7 commented:
FYI: Just like it was done for [WCAndroid](https://github.com/woocommerce/woocommerce-android/pull/13387#discussion_r1931952101) @wzieba, this is also 05222966e844d128ac469f44ef4e29c03d55ab9f ✅ for this project as well, everything is looking as expected:

- Before: [1563.65 MB](https://buildkite.com/automattic/wordpress-android/builds/20967#019498f1-bb31-4eba-8b4b-9a7bfda1bd43/23409-23421) cache size -> [4122](https://scans.gradle.com/s/x5tvcyzoybbia/performance/network-activity) network requests
- After: [1563.68 MB](https://buildkite.com/automattic/wordpress-android/builds/21026#0194ac93-2fe1-4820-9f83-38fa32d995d7/23407-23419) cache size -> [4108](https://scans.gradle.com/s/22quzciwex6co/performance/network-activity) network requests

PS: Note that although there exist some minor differences, but those are negligible and IMHO could be ignored.
